### ISSUE TITLE
exported MASKRCNN ONNX model cannot run:  Op (Slice) [ShapeInferenceError] Input axes has invalid data

### ISSUE THREAD

**Issue by qizhen816** [pytorch/vision#2308]
## 🐛 Bug
I exported my mask-rcnn model with resnet-101 as backbone using the most recently built torch and torchvision,  but cannot run by onnxruntime 1.3.0.
```
    img = cv2.resize(img,(1333,800))
    img = np.expand_dims(img,0)
    img = np.transpose(img,(0,3,1,2)).astype(np.float32)/255
    dummy_input1 = torch.from_numpy(img)
    model.eval()
    input_names = ["input"]
    torch.onnx.export(model, dummy_input1,
                      "traced_maskrcnn.onnx",
                      output_names=["boxes", "labels", "scores", "masks"],
                      # do_constant_folding=True,
                      verbose=True,
                      opset_version=11,
                      input_names=input_names)
```
Here is the export log:
```
C:\Users\Administrator\Anaconda3\lib\site-packages\torch\tensor.py:460: RuntimeWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).
  'incorrect results).', category=RuntimeWarning)
D:\Garage\maskvision\transforms.py:315: TracerWarning: torch.as_tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  mean = torch.as_tensor(self.image_mean, dtype=dtype, device=device)
D:\Garage\maskvision\transforms.py:316: TracerWarning: torch.as_tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  std = torch.as_tensor(self.image_std, dtype=dtype, device=device)
C:\Users\Administrator\Anaconda3\lib\site-packages\torchvision\models\detection\rpn.py:164: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  torch.tensor(image_size[1] // g[1], dtype=torch.int64, device=device)] for g in grid_sizes]
C:\Users\Administrator\Anaconda3\lib\site-packages\torchvision\ops\boxes.py:125: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  boxes_x = torch.min(boxes_x, torch.tensor(width, dtype=boxes.dtype, device=boxes.device))
C:\Users\Administrator\Anaconda3\lib\site-packages\torchvision\ops\boxes.py:127: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  boxes_y = torch.min(boxes_y, torch.tensor(height, dtype=boxes.dtype, device=boxes.device))
C:\Users\Administrator\Anaconda3\lib\site-packages\torchvision\ops\poolers.py:216: UserWarning: This overload of nonzero is deprecated:
	nonzero(Tensor input, *, Tensor out)
Consider using one of the following signatures instead:
	nonzero(Tensor input, *, bool as_tuple) (Triggered internally at  ..\torch\csrc\utils\python_arg_parser.cpp:761.)
  idx_in_level = torch.nonzero(levels == level).squeeze(1)
C:\Users\Administrator\Anaconda3\lib\site-packages\torchvision\models\detection\roi_heads.py:368: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  return torch.tensor(M + 2 * padding).to(torch.float32) / torch.tensor(M).to(torch.float32)
C:\Users\Administrator\Anaconda3\lib\site-packages\torch\onnx\symbolic_opset9.py:2225: UserWarning: Exporting aten::index operator of advanced indexing in opset 11 is achieved by combination of multiple ONNX operators, including Reshape, Transpose, Concat, and Gather. If indices include negative values, the exported graph will produce incorrect results.
  "If indices include negative values, the exported graph will produce incorrect results.")
```
The onnx model is generated after the above warnings, but onnxruntime throw an error:
`Node (Slice_1492) Op (Slice) [ShapeInferenceError] Input axes has invalid data`

**Comment by qizhen816** [comment]
Error disappeared after adding 
```
dynamic_axes={"input": [0, 1, 2, 3], "boxes": [0, 1], "labels": [0], "scores": [0], "masks": [0, 1, 2, 3]},
```
in `torch.onnx.export()` function, but new onnxruntime error occur:
`Non-zero status code returned while running Gather node. Name:'Gather_1542' Status Message: /mnt/onnxruntime/onnxruntime/core/providers/common.h:19 int64_t onnxruntime::HandleNegativeAxis(int64_t, int64_t) axis >= -tensor_rank && axis <= tensor_rank - 1 was false. axis 0 is not in valid range [-0,-1] `, which is similar to #2251, but passing identical batch size didn't help.

**Comment by neginraoof** [comment]
Thanks for reporting this. This is a known ONNX Runtime issue that I believe has been addressed very recently: https://github.com/microsoft/onnxruntime/pull/4185

I believe this fix will be released as part of ONNX Runtime nightly build (https://test.pypi.org/project/ort-nightly/) in a day or two. Will follow up.

**Comment by qizhen816** [comment]
@neginraoof Tried pulling and building the latest C++ version of ONNX Runtime, the `Op (Slice) ` error remains.

**Comment by qizhen816** [comment]
@neginraoof It's working now, thanks

**Comment by wangkd0** [comment]
Caused by: ai.onnxruntime.OrtException: Error code - ORT_RUNTIME_EXCEPTION - message: Non-zero status code returned while running Gather node. Name:'p2o.Gather.12' Status Message: C:\a\_work\1\s\onnxruntime\core/providers/common.h:32 onnxruntime::HandleNegativeAxis IsAxisInRange(axis, tensor_rank) was false. axis 0 is not in valid range [-0,-1]

I get this error when i use another photo
